centreon/tests/e2e/features/Dashboards/10-widget-metric-graph-configuration/index.ts

```diff
@@ -0,0 +1,395 @@
+    query: 'DELETE FROM dashboard'
+  });
+});
+
+after(() => {
+  cy.stopWebContainer();
+});
+
+Given(
+  "a dashboard in the dashboard administrator user's dashboard library",
+  () => {
+    cy.insertDashboard({ ...dashboards.default });
+    cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+    cy.getByLabel({
+      label: 'view',
+      tag: 'button'
+    })
+      .contains(dashboards.default.name)
+      .click();
+  }
+);
+
+When(
+  'the dashboard administrator user selects the option to add a new widget',
+  () => {
+    cy.get('*[class^="react-grid-layout"]').children().should('have.length', 0);
+    cy.getByTestId({ testId: 'edit_dashboard' }).click();
+    cy.getByTestId({ testId: 'AddIcon' }).click();
+  }
+);
+
+When('selects the widget type "Metrics graph"', () => {
+  cy.getByTestId({ testId: 'Widget type' }).click();
+  cy.contains('Metrics graph').click();
+});
+
+Then(
+  'configuration properties for the Metrics graph widget are displayed',
+  () => {
+    cy.contains('Widget properties').should('exist');
+    cy.getByLabel({ label: 'Title' }).should('exist');
+    cy.getByLabel({ label: 'RichTextEditor' }).should('exist');
+    cy.getByTestId({ testId: 'Time period' }).should('exist');
+  }
+);
+
+When(
+  'the dashboard administrator user selects a resource and a metric for the widget to report on',
+  () => {
+    cy.getByLabel({ label: 'Title' }).type(genericTextWidgets.default.title);
+    cy.getByLabel({ label: 'RichTextEditor' })
+      .eq(0)
+      .type(genericTextWidgets.default.description);
+    cy.getByTestId({ testId: 'Resource type' }).realClick();
+    cy.getByLabel({ label: 'Host Group' }).click();
+    cy.getByTestId({ testId: 'Select resource' }).click();
+    cy.contains('Linux-Servers').realClick();
+    cy.getByTestId({ testId: 'Select metric' }).click();
+    cy.contains('rta (ms) / Includes 1 resources').realClick();
+    cy.wait('@performanceData');
+  }
+);
+
+Then("a graph with a single bar is displayed in the widget's preview", () => {
+  cy.getByTestId({ testId: 'warning-line-200-tooltip' }).should('exist');
+  cy.getByTestId({ testId: 'critical-line-400-tooltip' }).should('exist');
+  cy.getByTestId({ testId: 'Min' }).should('exist');
+  cy.getByTestId({ testId: 'Max' }).should('exist');
+  cy.getByTestId({ testId: 'Avg' }).should('exist');
+});
+
+When(
+  'this bar represents the evolution of the selected metric over the default period of time',
+  () => {
+    cy.get('.visx-group.visx-axis-bottom text')
+      .last()
+      .invoke('text')
+      .should('match', /^(0?[1-9]|1[0-2]):[0-5][0-9] (AM|PM)$/);
+  }
+);
+
+When('the user saves the Metrics Graph widget', () => {
+  cy.getByTestId({ testId: 'confirm' }).click();
+});
+
+Then("the Metrics Graph widget is added to the dashboard's layout", () => {
+  cy.getByTestId({ testId: 'warning-line-200-tooltip' }).should('exist');
+  cy.getByTestId({ testId: 'critical-line-400-tooltip' }).should('exist');
```

Comment:
naive question here: is there another way for you to check that the widget exists besides the thresholds, which may or may not be there?

Reply:
unfortunately I haven't found an aria-label or test id that reflects the existence of a widget, so I've opted to make just these two assertions, like if these two elements exist, then the widget exists too